src/js/common/schemaform/review/ReviewChapters.jsx

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router';
+import ReviewCollapsibleChapter from './ReviewCollapsibleChapter';
+import {
+  createPageListByChapter,
+  expandArrayPages,
+  getActiveChapters,
+  getPageKeys
+} from '../helpers';
+import {
+  getReviewPageOpenChapters,
+  getViewedPages
+} from '../state/selectors';
+import { getActivePages } from '../../../../platform/forms/helpers';
+import {
+  closeReviewChapter,
+  openReviewChapter,
+  setData,
+  setEditMode,
+  setViewedPages,
+  uploadFile
+} from '../actions';
+import Scroll from 'react-scroll';
+
+const scroller = Scroll.scroller;
+class ReviewChapters extends React.Component {
+
+  componentDidMount() {
+    const {
+      formData,
+      pageList
+    } = this.props;
+    const viewedPages = new Set(getPageKeys(pageList, formData));
+    this.props.setViewedPages(viewedPages);
+  }
+
+  handleToggleChapter({ name, open, pageKeys }) {
+    if (open) {
+      this.props.closeReviewChapter(name, pageKeys);
+    } else {
+      this.props.openReviewChapter(name);
+      this.scrollToChapter(name);
+    }
+  }
+
+  scrollToChapter = (chapterKey) => {
+    scroller.scrollTo(`chapter${chapterKey}ScrollElement`, window.VetsGov.scroll || {
+      duration: 500,
+      delay: 2,
+      smooth: true,
+    });
+  }
+
+  handleEdit = (pageKey, editing, index = null) => {
+    const fullPageKey = `${pageKey}${index === null ? '' : index}`;
+    if (editing) {
+      this.props.setViewedPages([fullPageKey]);
+    }
+    this.props.setEditMode(pageKey, editing, index);
+  }
+
+  handleSetData = (...args) => {
+    this.props.setData(...args);
+    if (this.props.handleSetData) {
+      this.props.handleSetData();
+    }
+  }
+
+  render() {
+    const {
+      chapters,
+      form,
+      formContext,
+      setValid,
+      viewedPages
+    } = this.props;
+
+    return (
+      <div className="input-section">
+        <div>
+          {chapters.map(chapter => (
+            <ReviewCollapsibleChapter
+              expandedPages={chapter.expandedPages}
+              chapterFormConfig={chapter.formConfig}
+              chapterKey={chapter.name}
+              form={form}
```

Comment:
We don't necessarily need to do this here, but I think we could refactor ReviewCollapsibleChapter to take a formData and a pages prop, instead of passing the whole form state in.

Reply:
I added a TODO comment for this. I think there are a couple of things that we could do to this component to clean it up: 
- ditch the constructor and use anonymous function assignments instead of bind 
- inline the `pageContent` jsx or make it its own component that `ReviewChapters` composes (could move the prop creation that's in `ReviewCollapsibleChapter.render` into `ReviewChapters`'s `mapStateToProps`